tensorflow_addons/rnn/tests/cell_test.py

```diff
@@ -558,3 +558,64 @@ def test_esn_config():
     restored_cell = rnn_cell.ESNCell.from_config(config)
     restored_config = restored_cell.get_config()
     assert config == restored_config
+
+
+def test_zoneout():
+    units = 3
+    inputs = tf.constant([[[1.0, 2, 3, 4, 5]]])
+    initial_state = [tf.constant([[11.0, 12, 13]]), tf.constant([[21.0, 22, 23]])]
+
+    const_initializer = tf.constant_initializer(0.5)
+
+    # No zoneout. Testing against vanila LSTM.
+    lstm_layer = keras.layers.LSTM(
+        units,
+        kernel_initializer=const_initializer,
+        recurrent_initializer=const_initializer,
+        return_state=True,
+    )
+    lstm_output = lstm_layer(inputs, training=True)
+
+    zoneout_cell = rnn_cell.ZoneoutLSTMCell(
+        units,
+        kernel_initializer=const_initializer,
+        recurrent_initializer=const_initializer,
+    )
+    zoneout_layer = keras.layers.RNN(zoneout_cell, return_state=True)
+    zoneout_output = zoneout_layer(inputs, training=True)
+
+    np.testing.assert_allclose(lstm_output, zoneout_output)
+
+    # Total zoneout. Testing against initial state.
+    zoneout_cell = rnn_cell.ZoneoutLSTMCell(
+        units,
+        zoneout_h=1,
+        zoneout_c=1,
+        kernel_initializer=const_initializer,
+        recurrent_initializer=const_initializer,
+    )
+    zoneout_layer = keras.layers.RNN(zoneout_cell, return_state=True)
+    zoneout_output = zoneout_layer(inputs, initial_state=initial_state, training=True)
+
+    np.testing.assert_allclose(
+        [initial_state[0], initial_state[0], initial_state[1]], zoneout_output
+    )
+
+    # Inference. Expected to be the same as LSTM.
+    zoneout_output = zoneout_layer(inputs)
+
+    np.testing.assert_allclose(lstm_output, zoneout_output)
+
+
+def test_zoneout_config():
```

Comment:
You should test if you can create the cell using `from_config` function.
It could be something like this:
` 
restored_cell = rnn_cell.ZoneoutLSTMCell.from_config(config)
`
`
restored_config = restored_cell.get_config()
`
`
assert config == restored_config
`

Reply:
We do not override `from_config()` nor any of its helpers, why do we have to test it? When it could fail and how can we help if it fails? IMHO it's not a child class authors responsibility.

Reply:
I don't think the inherited method `from_config()` is going to fail, it just runs `cls(**config)`, where `cls` is `ZoneoutLSTMCell`, so it is calling the constructor. In this case, I'm sure it will not fail but it is better to have good tests implemented so that we avoid future mistakes while modifying this class (adding new arguments or changes in the `get_config`). 

A possible case when testing the `from_config` method can fail is if an argument from the constructor is not well defined in the `get_config` method.

Reply:
Remember, the only thing that may happen to a public class is an addition of an optional argument. Any other updates break backward compatibility and must be prohibited in the first place. At the end of the all, it's all about maintaining appropriate tests. Sorry, I personally see no reasons why to complicate the things.

Reply:
I was not talking about modifications of the LSTM layer but the child ZoneoutLSTM. 
I personally see no complication on adding those three lines that assure the complete process of initializing the recurrent cell from the config generated works fine. 
Apart from its utility, which I really believe in, It has been tested for every other recurrent cell in tensorflow_addons, so you can take consistency as another reason.